packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte

```diff
@@ -73,6 +73,35 @@
     | Types.Extension.IPaymentNetworkState<any>
     | undefined;
 
+  const checkStatus = (request: Types.IRequestDataWithEvents | null) => {
+    if (Number(request?.balance?.balance)! > 0) {
+      return request?.balance?.balance! >= request?.expectedAmount!
+        ? "Paid"
+        : "Partially Paid";
+    }
+
+    const eventStatus = {
+      create: "Created",
+      reject: "Rejected",
+      overdue: "Overdue",
+      cancel: "Canceled",
+    };
+
+    for (const [event, reqStatus] of Object.entries(eventStatus)) {
+      if (
+        request?.events?.some(
+          (e: { name?: string }) => e?.name?.toLowerCase() === event
+        )
+      ) {
+        return reqStatus;
+      }
+    }
+
+    return "Created";
+  };
+
+  let status = checkStatus(requestData || request);
```

Comment:
This logic correctly checks the balance first to determine whether the request is Paid or Partially Paid :+1: However, it should only return if Paid. If Partially Paid, it still needs to check the dueDate to determine if it's Overdue.

Please use [`request.state`](https://docs.request.network/learn-request-network/sdk-api-reference/request-client.js/irequestdatawithevents#types.requestlogic.state) to determine if the request is `pending`, `created`, `accepted`, or `canceled`.

If `state` is `canceled`, check the events to determine _who_ canceled. If the payee canceled, set the status to Canceled. If the payer canceled, set the status to Rejected

Finally, if status is Created, Accepted, or Partially Paid, check the dueDate and override the status to Overdue.

@sstefdev let me know if you need help crafting the if/else logic for this.

Reply:
@MantisClone I think that the events has overdue as well, I don't think that we need to check the dueDate, if we can access it directly, and see if the event is overdue, am I missing something?

Also the event can be rejected, that means that the payer rejected the invoice and if it is canceled, that means that the payee canceled the invoice.

Reply:
@MantisClone Overdue doesn't exist as an evnet, I fixed it and added check on our end to add that state.

I also abstracted the checkStatus function, so it is usable from the shared-utils.